obspy/taup/tau.py

```diff
@@ -663,3 +833,199 @@ def create_taup_model(model_name, output_dir, input_dir):
+        :type npoints: int
+        :param phase_list: List of phase names to plot.
+        :type phase_list: list of str, optional
+        :param model: string containing the model to use.
+        :type model: str
+        :param fig: Figure to plot in. If not given, a new figure instance
+            will be created.
+        :type fig: :class:`matplotlib.axes.Axes
+        :param ax: Axes to plot in. If not given, a new figure with an axes
+            will be created.
+        param show: Show the plot.
+        type show: bool
+        :type ax: :class:`matplotlib.Figure.figure`
+
+        :returns: ax
+        :rtype: :class:`matplotlib.axes.Axes`
+
+        .. rubric:: Example
+
+        >>> from obspy.taup import plot_travel_times
+        >>> import matplotlib.pyplot as plt
+        >>> fig, ax = plt.subplots()
+        >>> ax = plot_travel_times(source_depth=10, phase_list=['P','S','PP'],\
+        ax=ax, fig=fig)
+        There were 2 epicentral distances without an arrival
+
+        .. plot::
+
+            from obspy.taup import plot_travel_times
+            import matplotlib.pyplot as plt
+
+            fig, ax = plt.subplots()
+            ax = plot_travel_times(source_depth=10,
+                                   phase_list=['P','S','PP'], ax=ax, fig=fig)
+        """
+        import matplotlib.pyplot as plt
+
+        # compute the requested arrivals:
+        model = TauPyModel(model)
+
+        # a list of epicentral distances without a travel time, and a flag:
+        notimes = []
+        plotted = False
+
+        # calculate the arrival times and plot vs. epicentral distance:
+        degrees = np.linspace(min_degrees, max_degrees, npoints)
+        for degree in degrees:
+            try:
+                arrivals = model.get_ray_paths(source_depth, degree,
+                                               phase_list=phase_list)
+                ax = arrivals.plot_times(phase_list=phase_list, show=False,
+                                         ax=ax)
+                plotted = True
+            except ValueError as err:
+                notimes.append(degree)
+                pass
+
+        if plotted:
+            print(("There were {} epicentral distances "
+                   "without an arrival").format(len(notimes)))
+        else:
+            raise ValueError("No arrival times to plot.")
+
+        # merge all arrival labels of a certain phase:
+        handles, labels = ax.get_legend_handles_labels()
+        labels, ids = np.unique(labels, return_index=True)
+        handles = [handles[i] for i in ids]
+        ax.legend(handles, labels, loc=2, numpoints=1)
+
+        if show:
+            plt.show()
+        return ax
+
+
+def plot_ray_paths(source_depth, min_degrees=0, max_degrees=360, npoints=10,
+                   plot_type='spherical', phase_list=['P', 'S', 'PP'],
+                   model='iasp91', plot_all=True, legend=False,
+                   label_arrivals=False, fig=None, show=True, ax=None):
```

Comment:
@kaspervanwijk I think there's a caveat about `plot_type`/`ax` that is not covered yet. What's your take on the following point?

 - if `ax` is specified, then depending on what kinds of plots have been done before this could get a garbled mixed plot with both spherical and cartesian plots overlain? I think we should raise an `ValueError` if `plot_type='spherical'` and provided axes (if any) `isinstance(ax, matplotlib.projections.polar.PolarAxes)` (and vice versa for polar plot)?


Just wanted to hear if this sounds about right, I'm already at some minor changes to finalize this PR and can do it.

Reply:
Not sure if I follow. Do you mean that if someone calls this helper function in a script, and had already plotted something (maybe in cartesian coordinates), before a plt.show() and then this would send a polar plot to that instance?

Reply:
Yep, something like that..

Reply:
In any case, I do my final change (push them maybe in an hour or so) and then you can have a look and comment on them later.

Reply:
Ah.. I see, that wasn't in your changes, really.. it's just in the diff because it was moved over from the old `plot` routine to `plot_rays`..